Description: Upstream changes introduced in version 1.0.2-1+~svn14032
 This patch has been created by dpkg-source during the package build.
 Here's the last changelog entry, hopefully it gives details on why
 those changes were made:
 .
 mlpack (1.0.2-1+~svn14032) UNRELEASED; urgency=low
 .
   * Initial release (Closes: #100)
 .
 The person named in the Author field signed this changelog entry.
Author: Sterling Lewis Peet <sterling.peet@gatech.edu>
Bug-Debian: http://bugs.debian.org/100

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- /dev/null
+++ mlpack-1.0.2/rpm/range_search_install.patch
@@ -0,0 +1,7 @@
+--- mlpack-1.0.1/src/mlpack/methods/range_search/CMakeLists.txt 	2012-03-07 23:56:53.000000000 -0500
++++ mlpack-1.0.1/src/mlpack/methods/range_search/CMakeLists.txt.new	2012-03-07 23:56:31.000000000 -0500
+@@ -22,3 +22,4 @@
+ target_link_libraries(range_search
+   mlpack
+ )
++install(TARGETS range_search RUNTIME DESTINATION bin)
--- /dev/null
+++ mlpack-1.0.2/rpm/mlpack.spec
@@ -0,0 +1,198 @@
+Name:           mlpack
+Version:        1.0.1
+Release:        3%{?dist}
+Summary:        Scalable, fast C++ machine learning library
+
+Group:          System Environment/Libraries
+License:        LGPLv3+
+URL:            http://www.mlpack.org
+Source0:        http://www.mlpack.org/files/%{name}-%{version}.tar.gz
+
+# CMake by default doesn't provide support for specifying libdir on install
+# (argh!) and this has to be done by hand in the CMake configuration.  This'll
+# be fixed by mlpack 1.0.2.
+Patch0:         cmake_libdir.patch
+
+# Oversight means that the rule which says "install range_search to bin/" didn't
+# get there.
+Patch1:         range_search_install.patch
+
+# The GetKernelMatrix patch fixes a bug where a function was called before
+# it was ever declared.  This isn't a problem in certain GCC settings, but
+# in the Fedora setting it throws an error (which is easily fixed by that
+# patch).  Fixed upstream in trunk (not 1.0.1).
+Patch2:         getkernelmatrix_definition.patch
+
+# More recent Armadillo versions require specific calls to arma::as_scalar().
+# Fixed upstream in trunk (not 1.0.1).
+Patch3:         fully_qualified_as_scalar.patch
+
+# The hacked-in sparse support for Armadillo in mlpack 1.0.1 does not work with
+# recent Armadillo versions.  That support becomes irrelevant with the
+# impending 3.6 release so the issue is not fixed in trunk due to its pending
+# obsolescence.
+Patch4:         no_sparse_tests.patch
+
+BuildRoot:      %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
+
+BuildRequires:  cmake >= 2.8.0
+BuildRequires:  armadillo-devel >= 2.4.0
+BuildRequires:  libxml2-devel
+BuildRequires:  boost-devel, boost-program-options, boost-math
+BuildRequires:  lapack-devel
+# For generating man pages (CMake configuration takes care of this assuming
+# txt2man is installed).  It is possible that we could just add all the man
+# pages, generated offline, as a patch to this SRPM, but txt2man seems to exist
+# in repos.
+BuildRequires:  txt2man
+
+# 2.4.0 precipitated an internal Armadillo switch from 'u32' to 'uword' and the
+# Armadillo extensions in mlpack depend on the use of 'uword' which is
+# undefined <= 2.4.0.
+Requires:       armadillo >= 2.4.0
+Requires:       libxml2
+Requires:       boost, boost-program-options, boost-math
+Requires:       lapack
+
+%description
+mlpack is a C++ machine learning library with emphasis on scalability, speed,
+and ease-of-use. Its aim is to make machine learning possible for novice users
+by means of a simple, consistent API, while simultaneously exploiting C++
+language features to provide maximum performance and maximum flexibility for
+expert users. mlpack outperforms competing machine learning libraries by large
+margins.
+
+# Executables.
+%package bin
+Summary:        Command-line executables for mlpack (machine learning library)
+Group:          Applications/Engineering
+Requires:       %{name}%{_isa} = %{version}-%{release}
+
+%description bin
+mlpack is a C++ machine learning library with emphasis on scalability, speed,
+and ease-of-use. Its aim is to make machine learning possible for novice users
+by means of a simple, consistent API, while simultaneously exploiting C++
+language features to provide maximum performance and maximum flexibility for
+expert users. mlpack outperforms competing machine learning libraries by large
+margins.  This package provides the command-line executables which run mlpack
+methods and related documentation.
+
+# Development headers.
+%package devel
+Summary:        Development headers for mlpack (C++ machine learning library)
+Group:          System Environment/Libraries
+Requires:       %{name} = %{version}-%{release}
+Requires:       armadillo-devel >= 2.4.0
+Requires:       boost-devel, boost-program-options, boost-math
+Requires:       libxml2-devel
+Requires:       lapack-devel
+
+%description devel
+mlpack is a C++ machine learning library with emphasis on scalability, speed,
+and ease-of-use. Its aim is to make machine learning possible for novice users
+by means of a simple, consistent API, while simultaneously exploiting C++
+language features to provide maximum performance and maximum flexibility for
+expert users. mlpack outperforms competing machine learning libraries by large
+margins.  This package provides the headers to compile applications against
+mlpack.
+
+
+%prep
+%setup -q
+%patch0 -p1
+%patch1 -p1
+%patch2 -p1
+%patch3 -p1
+%patch4 -p1
+
+
+%build
+%{cmake} -D LIBDIR=%{_libdir} -D DEBUG=OFF -D PROFILE=OFF .
+make %{?_smp_mflags}
+
+
+%install
+rm -rf $RPM_BUILD_ROOT
+make install DESTDIR=$RPM_BUILD_ROOT
+
+# Rename executables to mlpack_* to avoid possible naming collisions.  Upstream
+# bug report http://trac.research.cc.gatech.edu/fastlab/ticket/236.
+for i in `ls $RPM_BUILD_ROOT/%{_bindir}`
+do
+  %{__mv} $RPM_BUILD_ROOT/%{_bindir}/$i $RPM_BUILD_ROOT/%{_bindir}/mlpack_$i;
+done
+for i in `ls $RPM_BUILD_ROOT/%{_mandir}/man1`
+do
+  %{__mv} $RPM_BUILD_ROOT/%{_mandir}/man1/$i $RPM_BUILD_ROOT/%{_mandir}/man1/mlpack_$i;
+done
+
+%clean
+rm -rf $RPM_BUILD_ROOT
+
+%post -p /sbin/ldconfig
+
+%postun -p /sbin/ldconfig
+
+
+%files
+%defattr(-,root,root,-)
+%{_libdir}/libmlpack.so.1
+%{_libdir}/libmlpack.so.1.0.1
+
+%files bin
+%defattr(-,root,root,-)
+%{_bindir}/mlpack_allknn
+%{_bindir}/mlpack_allkfn
+%{_bindir}/mlpack_emst
+%{_bindir}/mlpack_gmm
+%{_bindir}/mlpack_hmm_generate
+%{_bindir}/mlpack_hmm_loglik
+%{_bindir}/mlpack_hmm_train
+%{_bindir}/mlpack_hmm_viterbi
+%{_bindir}/mlpack_kernel_pca
+%{_bindir}/mlpack_kmeans
+%{_bindir}/mlpack_lars
+%{_bindir}/mlpack_linear_regression
+%{_bindir}/mlpack_local_coordinate_coding
+%{_bindir}/mlpack_nbc
+%{_bindir}/mlpack_nca
+%{_bindir}/mlpack_pca
+%{_bindir}/mlpack_radical
+%{_bindir}/mlpack_range_search
+%{_bindir}/mlpack_sparse_coding
+%{_mandir}/man1/mlpack_allknn.1*
+%{_mandir}/man1/mlpack_allkfn.1*
+%{_mandir}/man1/mlpack_emst.1*
+%{_mandir}/man1/mlpack_gmm.1*
+%{_mandir}/man1/mlpack_hmm_generate.1*
+%{_mandir}/man1/mlpack_hmm_loglik.1*
+%{_mandir}/man1/mlpack_hmm_train.1*
+%{_mandir}/man1/mlpack_hmm_viterbi.1*
+%{_mandir}/man1/mlpack_kernel_pca.1*
+%{_mandir}/man1/mlpack_kmeans.1*
+%{_mandir}/man1/mlpack_lars.1*
+%{_mandir}/man1/mlpack_linear_regression.1*
+%{_mandir}/man1/mlpack_local_coordinate_coding.1*
+%{_mandir}/man1/mlpack_nbc.1*
+%{_mandir}/man1/mlpack_nca.1*
+%{_mandir}/man1/mlpack_pca.1*
+%{_mandir}/man1/mlpack_radical.1*
+%{_mandir}/man1/mlpack_range_search.1*
+%{_mandir}/man1/mlpack_sparse_coding.1*
+
+%files devel
+%defattr(-,root,root,-)
+%{_libdir}/libmlpack.so
+%{_includedir}/mlpack/
+
+%changelog
+* Sun Jul 29 2012 Ryan Curtin <gth671b@mail.gatech.edu> - 1.0.1-3
+- Fix group names for packages.
+- Comment patches more verbosely.
+- Rename exectuables to mlpack_* to avoid possible naming conflicts.
+
+* Thu Jul 21 2012 Sterling Lewis Peet <sterling.peet@gatech.edu> - 1.0.1-2
+- Include GetKernelMatrix patch so that mlpack builds using fedora flags.
+
+* Thu Mar 08 2012 Ryan Curtin <gth671b@mail.gatech.edu> - 1.0.1-1
+- Initial packaging of mlpack.
--- /dev/null
+++ mlpack-1.0.2/rpm/getkernelmatrix_definition.patch
@@ -0,0 +1,36 @@
+Index: mlpack-1.0.1/src/mlpack/methods/kernel_pca/kernel_pca.hpp
+===================================================================
+--- mlpack-1.0.1/src/mlpack/methods/kernel_pca/kernel_pca.hpp	(revision 13174)
++++ mlpack-1.0.1/src/mlpack/methods/kernel_pca/kernel_pca.hpp.new	(working copy)
+@@ -88,6 +88,9 @@
+ 
+ }; // class KernelPCA
+ 
++template <typename KernelType>
++arma::mat GetKernelMatrix(KernelType kernel, arma::mat transData);
++
+ }; // namespace kpca
+ }; // namespace mlpack
+ 
+Index: mlpack-1.0.1/src/mlpack/methods/kernel_pca/kernel_pca_impl.hpp
+===================================================================
+--- mlpack-1.0.1/src/mlpack/methods/kernel_pca/kernel_pca_impl.hpp	(revision 13174)
++++ mlpack-1.0.1/src/mlpack/methods/kernel_pca/kernel_pca_impl.hpp.new	(working copy)
+@@ -109,9 +109,6 @@
+     data.shed_rows(newDimension, data.n_rows - 1);
+ }
+ 
+-}; // namespace mlpack
+-}; // namespace kpca
+-
+ template <typename KernelType>
+ arma::mat GetKernelMatrix(KernelType kernel, arma::mat transData)
+ {
+@@ -130,4 +127,7 @@
+   return kernelMat;
+ }
+ 
++}; // namespace mlpack
++}; // namespace kpca
++
+ #endif
--- /dev/null
+++ mlpack-1.0.2/rpm/cmake_libdir.patch
@@ -0,0 +1,28 @@
+--- mlpack-1.0.1/CMakeLists.txt 	2012-03-07 23:35:40.000000000 -0500
++++ mlpack-1.0.1/CMakeLists.txt.new	2012-03-07 23:37:32.000000000 -0500
+@@ -75,6 +75,12 @@
+ option(PROFILE "Compile with profiling information" ON)
+ option(ARMA_EXTRA_DEBUG "Compile with extra Armadillo debugging symbols." OFF)
+
++if(LIBDIR)
++  message(STATUS "Setting library install directory to ${LIBDIR}.")
++else(LIBDIR)
++  set(LIBDIR "lib") # Default.
++endif()
++
+ # This is as of yet unused.
+ #option(PGO "Use profile-guided optimization if not a debug build" ON)
+
+--- mlpack-1.0.1/src/mlpack/CMakeLists.txt	2012-03-07 23:35:47.000000000 -0500
++++ mlpack-1.0.1/src/mlpack/CMakeLists.txt.new	2012-03-07 23:36:03.000000000 -0500
+@@ -55,8 +55,8 @@
+ # be entered...
+ install(TARGETS mlpack
+   RUNTIME DESTINATION bin
+-  LIBRARY DESTINATION lib
+-  ARCHIVE DESTINATION lib)
++  LIBRARY DESTINATION ${LIBDIR}
++  ARCHIVE DESTINATION ${LIBDIR})
+
+ # For 'make test'.
+ add_custom_target(test
--- /dev/null
+++ mlpack-1.0.2/rpm/armadillo.spec
@@ -0,0 +1,238 @@
+Name:           armadillo
+Version:        3.0.4
+Release:        1%{?dist}
+Summary:        Fast C++ matrix library with interfaces to LAPACK and ATLAS
+
+Group:          Development/Libraries
+License:        LGPLv3+
+URL:            http://arma.sourceforge.net/
+Source:         http://sourceforge.net/projects/arma/files/%{name}-%{version}.tar.gz
+BuildRoot:      %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
+BuildRequires:  cmake, boost-devel, blas-devel, lapack-devel, atlas-devel
+
+%description
+Armadillo is a C++ linear algebra library (matrix maths)
+aiming towards a good balance between speed and ease of use.
+Integer, floating point and complex numbers are supported,
+as well as a subset of trigonometric and statistics functions.
+Various matrix decompositions are provided through optional
+integration with LAPACK and ATLAS libraries.
+A delayed evaluation approach is employed (during compile time)
+to combine several operations into one and reduce (or eliminate) 
+the need for temporaries. This is accomplished through recursive
+templates and template meta-programming.
+This library is useful if C++ has been decided as the language
+of choice (due to speed and/or integration capabilities), rather
+than another language like Matlab or Octave.
+
+
+%package devel
+Summary:        Development headers and documentation for the Armadillo C++ library
+Group:          Development/Libraries
+Requires:       %{name} = %{version}-%{release}
+Requires:       boost-devel, blas-devel, lapack-devel, atlas-devel, libstdc++-devel
+
+# The header files of Armadillo include some Boost and ATLAS header files,
+# delivered within the boost-devel and atlas-devel sub-packages, respectively.
+# However, since there is no explicit dependency on Boost or ATLAS libraries
+# (most of Boost is delivered as header files only), the RPM building process 
+# does not detect these dependencies.  These dependencies must therefore be 
+# added manually.
+
+%description devel
+This package contains files necessary for development using the
+Armadillo C++ library. It contains header files, example programs,
+and user documentation (reference guide).
+
+
+%prep
+%setup -q
+
+# convert DOS end-of-line to UNIX end-of-line
+
+for file in README.txt; do
+  sed 's/\r//' $file >$file.new && \
+  touch -r $file $file.new && \
+  mv $file.new $file
+done
+
+%build
+%{cmake}
+%{__make} VERBOSE=1 %{?_smp_mflags}
+
+
+%install
+rm -rf $RPM_BUILD_ROOT
+%{__make} install DESTDIR=$RPM_BUILD_ROOT
+rm -rf   $RPM_BUILD_ROOT/%{_docdir}/%{name}-%{version}/
+mkdir -p $RPM_BUILD_ROOT/%{_docdir}/%{name}-%{version}/
+rm -f examples/Makefile.cmake
+rm -rf examples/example1_win32
+rm -rf examples/example2_win32
+rm -rf examples/lib_win32
+cp -r LICENSE.txt licenses README.txt index.html examples docs $RPM_BUILD_ROOT/%{_docdir}/%{name}-%{version}/
+
+%clean
+rm -rf $RPM_BUILD_ROOT
+
+
+%post -p /sbin/ldconfig
+
+%postun -p /sbin/ldconfig
+
+
+%files
+%defattr(-,root,root,-)
+%{_libdir}/*.so.*
+%dir %{_docdir}/%{name}-%{version}/
+%doc %{_docdir}/%{name}-%{version}/LICENSE.txt
+%doc %{_docdir}/%{name}-%{version}/licenses/
+
+%files devel
+%defattr(-,root,root,-)
+%{_libdir}/*.so
+%{_includedir}/armadillo
+%{_includedir}/armadillo_bits/
+%doc %{_docdir}/%{name}-%{version}/README.txt
+%doc %{_docdir}/%{name}-%{version}/index.html
+%doc %{_docdir}/%{name}-%{version}/examples/
+%doc %{_docdir}/%{name}-%{version}/docs/
+%{_datadir}/Armadillo/
+
+%changelog
+* Wed Jul 18 2012 Ryan Curtin <ryan@igglybob.com> - 3.0.4-1
+- spec updated for Armadillo 3.0.4
+
+* Fri Jul 06 2012 Sterling Peet - 2.4.0-1
+- spec updated for Armadillo 2.4.0
+
+* Thu Jan 12 2012 Fedora Release Engineering <rel-eng@lists.fedoraproject.org> - 2.2.3-2
+- Rebuilt for https://fedoraproject.org/wiki/Fedora_17_Mass_Rebuild
+
+* Thu Sep 15 2011 Conrad Sanderson - 2.2.3-1
+- spec updated for Armadillo 2.2.3
+
+* Mon Apr 18 2011 Conrad Sanderson - 1.2.0-1
+- spec updated for Armadillo 1.2.0
+
+* Mon Nov 15 2010 Conrad Sanderson - 1.0.0-1
+- spec updated for Armadillo 1.0.0
+
+* Thu Oct 14 2010 Conrad Sanderson - 0.9.90-1
+- spec updated for Armadillo 0.9.90
+
+* Tue Sep 21 2010 Conrad Sanderson - 0.9.80-1
+- spec updated for Armadillo 0.9.80
+
+* Wed Sep 01 2010 Conrad Sanderson - 0.9.70-1
+- spec updated for Armadillo 0.9.70
+
+* Wed Aug 04 2010 Conrad Sanderson - 0.9.60-1
+- spec updated for Armadillo 0.9.60
+
+* Wed Jul 14 2010 Conrad Sanderson - 0.9.52-1
+- spec updated for Armadillo 0.9.52
+
+* Wed Jul 07 2010 Conrad Sanderson - 0.9.50-1
+- spec updated for Armadillo 0.9.50
+
+* Wed Jun 02 2010 Conrad Sanderson - 0.9.10-1
+- spec updated for Armadillo 0.9.10
+
+* Fri May 14 2010 Conrad Sanderson - 0.9.8-1
+- spec updated for Armadillo 0.9.8
+
+* Tue Apr 28 2010 Conrad Sanderson - 0.9.6-1
+- spec updated for Armadillo 0.9.6
+
+* Tue Mar 16 2010 Conrad Sanderson - 0.9.4-1
+- spec updated for Armadillo 0.9.4
+
+* Tue Mar 02 2010 Conrad Sanderson - 0.9.2-2
+- added explicit dependencies to the devel package
+
+* Tue Mar 02 2010 Conrad Sanderson - 0.9.2-1
+- spec updated for Armadillo 0.9.2
+
+* Fri Feb 05 2010 Conrad Sanderson - 0.9.0-1
+- spec updated for Armadillo 0.9.0
+
+* Mon Jan 27 2010 Conrad Sanderson - 0.8.2-1
+- spec updated for Armadillo 0.8.2
+
+* Mon Dec 14 2009 Conrad Sanderson - 0.8.0-1
+- spec updated for Armadillo 0.8.0
+
+* Fri Oct 23 2009 Conrad Sanderson - 0.7.2-1
+- spec updated for Armadillo 0.7.2
+
+* Mon Oct 05 2009 Conrad Sanderson - 0.7.0-1
+- spec updated for Armadillo 0.7.0
+
+* Fri Jul 24 2009 Fedora Release Engineering <rel-eng@lists.fedoraproject.org> - 0.6.12-3
+- Rebuilt for https://fedoraproject.org/wiki/Fedora_12_Mass_Rebuild
+
+* Wed Jul 06 2009  Conrad Sanderson - 0.6.12-2
+- added conversion of DOS end-of-line to UNIX end-of-line for README.txt
+
+* Wed Jun 22 2009  Conrad Sanderson - 0.6.12-1
+- spec updated for Armadillo 0.6.12
+
+* Wed Jun 15 2009  Conrad Sanderson - 0.6.11-8
+- cleanup of dependencies
+- explanation as to why boost-devel and atlas-devel are required by armadillo-devel
+
+* Wed Jun 09 2009  Conrad Sanderson - 0.6.11-7
+- explicit declaration of doc directory in the main package
+- explicitly marked doc files in both packages
+
+* Wed Jun 09 2009  Conrad Sanderson - 0.6.11-6
+- removed symlinks
+- placed all documentation and license files into one directory that is shared by both packages
+
+* Wed Jun 09 2009  Conrad Sanderson - 0.6.11-5
+- added symlinks to LICENSE.txt and licenses in the devel package
+
+* Wed Jun 08 2009  Conrad Sanderson - 0.6.11-4
+- added LICENSE.txt to the main package
+
+* Wed May 22 2009  Conrad Sanderson - 0.6.11-3
+- using cmake macro instead of directly calling cmake
+
+* Wed May 21 2009  Conrad Sanderson - 0.6.11-2
+- moved all text files to devel package to retain consistency with the layout in the original .tar.gz
+
+* Wed May 08 2009  Conrad Sanderson - 0.6.10-2
+- Removed several explicit build dependencies that are provided by default in Fedora
+- Simplified handling of doc files
+
+* Wed May 02 2009  Conrad Sanderson - 0.6.10-1
+- Updated spec file for Armadillo 0.6.10
+
+* Wed Apr 02 2009  Conrad Sanderson
+- Updated list of files in 0.6.7 release
+
+* Wed Apr 02 2009  Conrad Sanderson
+- Updated description
+
+* Wed Mar 24 2009  Conrad Sanderson
+- Added explicit dependence on libstdc++-devel
+
+* Wed Mar 17 2009  Conrad Sanderson
+- Simplified specification of directories
+- Removed library packages specified by "Requires", as library dependencies are detected automatically
+
+* Wed Mar 12 2009  Conrad Sanderson
+- Modified to generate separate devel package (subsumes previous doc package)
+- Removed redundant packages specified by "BuildRequires"
+- Added CMake installation prefixes to allow for x86_64
+
+* Wed Feb  4 2009  Conrad Sanderson
+- Modified to generate separate doc package
+
+* Thu Jan 28 2009  Conrad Sanderson
+- Added argument to cmake: -DCMAKE_INSTALL_PREFIX=/usr 
+
+* Thu Jan 22 2009  Conrad Sanderson
+- Initial spec file prepared
+
